File: src/python/240326.py
 def firstMissingPositive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums=sorted(set(nums))
        c=0
        for i in nums:
            if i > 0 :
                c +=1
                if i != c:
                    return c
        return c+1

nums = [1,2,4]
res = firstMissingPositive(nums)
print(res)

File: src/php/240326.php
<?php
$nums = [2,3,4,5,6];
echo firstMissingPositive($nums);
function firstMissingPositive($nums) {
    // sort($nums);
    // $high = $nums[count($nums)-1];
    // if($high < 1)
    // return 1;
    $nums = array_flip($nums);
    for($i=1; $i<=count($nums)+1; $i++) {
        if (!isset($nums[$i])) return $i;
    }
}